test/release/examples/primers/loops.chpl

```diff
@@ -0,0 +1,950 @@
+  the :ref:`Parallel Iterators Primer <primers-parIters>`.
+
+  A parallel iterator can create as many tasks as it wants, and can
+  specify where they should run.  By convention, most will create as
+  many tasks as are appropriate for the hardware parallelism that the
+  loop iterand targets.  For example, the default parallel iterator
+  for a range, local domain, or local array typically implements its
+  iterations using a number of tasks equal to the number of local
+  processor cores that are available, since those data structures are
+  stored on a single locale.  In contrast, the default parallel
+  iterator for a distributed domain or array will typically implement
+  the iterations using all of the available processor cores on all of
+  the locales that own a subset of the domain's indices or array's
+  elements.
+
+  It is guaranteed that all tasks created by the parallel iterator to
+  run the loop's iterations will complete before the original task
+  encountering the forall-loop proceeds.  Logically, you can think of
+  there as being a logical *join* operation on all tasks that are
+  helping to implement the forall-loop.
+
+  Looking at some simple examples, when run on a k-core processor, for
+  large enough values of ``n``, each of the following loops will
+  typically use ``k`` tasks to implement the loop's iterations in
+  parallel:
+
+*/
+
+  config const n = 1000;
+
+  var B: [1..n] real;
+
+
+  forall i in 1..n do
+    B[i] = i: real;
+
+  writeln("After the forall loop over a range, B is: ", B);
+
+
+  forall i in B.domain do
+    B[i] = A[i % A.size];
+
+  writeln("After the forall loop over a domain, B is: ", B);
+
+
+  forall b in B do
+    b = -b;
+
+  writeln("After the forall loop over an array, B is: ", B);
+
+/*
+
+  Similarly, if we iterate over a domain or array that's distributed
+  across multiple locales with ``k`` cores each, each locale will tend
+  to use its ``k`` cores to iterate over the subset of the domain or
+  array that it owns locally:
+  
+*/
+
+  use BlockDist;
+
+  const BlockDom = Block.createDomain({1..n, 1..n});
+  var C: [BlockDom] real;
+
+  forall (i,j) in BlockDom do
+    C[i,j] = (100 * here.id) + i + j/1000.0;
+
+  writeln("After the forall loop over a distributed domain, C is:\n",C);
+
+  forall c in C do
+    c *= 2;
+
+  writeln("After the forall loop over a distributed array, C is:\n", C);
+
+/*
+
+  Because forall-loops invoke parallel iterators, the tasks they
+  create and where they run are not defined by the Chapel language,
+  but by the iterators themselves.  Any type supporting parallel
+  iteration should describe how its parallel iterators work as part of
+  its user-facing documentation.
+
+  Square Bracket Loops
+  --------------------
+
+  A third data-parallel loop form uses square brackets to define the
+  loop instead of the ``foreach`` or ``forall`` keywords.  For
+  example, such a loop may look like:
+
+*/
+  
+  [c in C] c -= 0.001;
```

Comment:
It would be really good to have an example with a range since that is going to be more familiar to people then using arrays like this.